src/org/omegat/filters2/text/bundles/ResourceBundleFilter.java

```diff
@@ -263,8 +296,10 @@ private String toAscii(String text, EscapeMode mode) {
                 result.append("\\=");
             } else if (mode == EscapeMode.KEY && cp == ':') {
                 result.append("\\:");
-            } else if ((cp >= 32 && cp < 127)
-                    || charsetEncoder.canEncode(text.substring(i, i + Character.charCount(cp)))) {
+            } else if (cp >= 32 && cp < 127) {
+                result.appendCodePoint(cp);
+            } else if (!forceTargetEscape
+                    && charsetEncoder.canEncode(text.substring(i, i + Character.charCount(cp)))) {
                 result.appendCodePoint(cp);
```

Comment:
### Change 2.

We set default to be UTF-8 so `canEncode` might return TRUE but we introduce `forceTargetEscape` flag that is true in default force the behavior  as same as  OmegaT 5.7.1.